From 727c347bda3b14715a3f35254ed5cc33d8538926 Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Thu, 21 Jan 2016 12:51:58 +0000
Subject: [PATCH 09/13] Revert "Disable LPM Support in BOS descriptor."

This reverts commit 091dbc6e30589fe2aff8f5d306acd06e1f9a82e2.

A possible solution for the LPM problem has been introduced
(https://android.intel.com/#/c/451096), hence the patch for
disabling the LPM_SUPPORT in the dwc3 driver needs to be
reverted.

Change-Id: I508d9110858948a55f81170aae45f8bfe5a5d2c0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6490
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/460822
---
 drivers/usb/gadget/composite.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 1f8c822..fe076e1 100644
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -528,7 +528,7 @@ static int bos_desc(struct usb_composite_dev *cdev)
 	usb_ext->bLength = USB_DT_USB_EXT_CAP_SIZE;
 	usb_ext->bDescriptorType = USB_DT_DEVICE_CAPABILITY;
 	usb_ext->bDevCapabilityType = USB_CAP_TYPE_EXT;
-	usb_ext->bmAttributes = 0;
+	usb_ext->bmAttributes = cpu_to_le32(USB_LPM_SUPPORT | USB_BESL_SUPPORT);
 
 	/*
 	 * The Superspeed USB Capability descriptor shall be implemented by all
-- 
1.9.1

